todo-app/src/EditBoard.js

```diff
@@ -0,0 +1,67 @@
+import { useState, useEffect } from "react";
+import "./EditBoard.css";
+
+export function EditBoard({
+  todos,
+  onAddTodos,
+  onUpdateTodos,
+  onDeleteTodo,
+  selectedTodoId,
+}) {
+  const targetTodo = todos.find((todo) => todo.id === selectedTodoId);
+  const [editingTodo, setEditingTodo] = useState({ id: 0, text: "" });
+  const [noInput, setNoInput] = useState(false);
+
+  useEffect(() => {
+    setEditingTodo(targetTodo ? targetTodo : { id: 0, text: "" });
+    setNoInput(false);
+  }, [targetTodo]);
+
+  function updateTextarea(e) {
+    setEditingTodo({ ...editingTodo, text: e.target.value });
+    setNoInput(e.target.value === "" ? true : false);
+  }
```

Comment:
[imo]
こんな感じの方がわかりやすそう？

```suggestion
  function updateTextarea(e) {
    const value = e.target.value;
    setEditingTodo({ ...editingTodo, text: value });
    setNoInput(!value);
  }
```